rasa/utils/io.py

```diff
@@ -212,3 +213,26 @@ def json_pickle(file_name: Union[Text, Path], obj: Any) -> None:
     jsonpickle_numpy.register_handlers()
 
     rasa.shared.utils.io.write_text_file(jsonpickle.dumps(obj), file_name)
+
+
+def is_valid_remote_url(url: Text) -> bool:
+    """Check whether the url specified is a well formed one.
+
+    Regex taken from https://stackoverflow.com/a/7160778/3001665
+
+    Args:
+        url: Remote URL to validate
+
+    Returns: Boolean for validity
+    """
+    regex = re.compile(
```

Comment:
Can we use something like https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlparse and check if the protocol is given and `http(s)`? seems more robust in my opinion.

Reply:
I tried using it and looking like the check would be something like if `scheme` and `netloc` are defined then it is a valid URL but it would fail in a simple case like `https://a/b/c` where `a` is extracted as `netloc` and `b/c` as path. Ideally it should have been `a/b/c` as path and then `netloc` just empty. Not sure if that is more robust that a regex.

Reply:
Makes sense. Didn't know this function existed. I have replaced the original regex here with the new regex. Handles other cases like IP, ports, etc.